File: Euler38.java
public class Euler38 {
  public static void main(String args[]){
    int largest = 0;
    final int LIMIT = 1000000000;
    for(int i = 1; i < LIMIT/2; i++){
      int num = 1;
      int n = 2;
      boolean run = true;
      while(run){
        long test = concatFunc(i,n);
        if(test>LIMIT){
          run = false;
        }
        else{
          num=(int)test;
          n++;
        }
      }

      if(num>largest && pandigital(num)){
        largest = num;
        System.out.println(largest);
      }
    }
    System.out.println(largest);
  }

  public static long concatFunc(int num, int nVal){
    String result = ""+num;
    for(int j = 2; j <= nVal; j++){
      int addTo = j*num;
      result=""+result+addTo;
    }
    return Long.parseLong(result);
  }

  public static boolean pandigital(int num){
    String val = ""+num;
    if(val.length() < 8 || val.length() > 8){
      return false;
    }
    for (int k = 1; k <= 9; k++) {
      if(!val.contains(""+k)){
        return false;
      }
    }
    return true;
  }

}

File: Euler38.py
def f(num, max):
    result = ''
    for i in range(1, max+1):
        result += str(num*i)
    return result


def isPan(num):
    if len(num) != 9:
        return False
    for i in range(1, 10):
        if str(i) not in num:
            return False
    return True


run = 9765

while run > 1:
    if isPan(f(run, 2)):
        print(f(run, 2))
        break
    else:
        run -= 1
